src/plugins/bbsync/sync_thread.cpp

```diff
@@ -124,8 +138,16 @@ BlackBoardSynchronizationThread::finalize()
 void
 BlackBoardSynchronizationThread::loop()
 {
+	if (connect_failed_since_ == 4) {
```

Comment:
A few things. First, the value `4` here comes out of the blue and should be something that is configurable, same goes for the value `5` below.
Second, you never actually set the aliveness back to true, once the connection is re-established, right?